core/src/main/java/org/apache/spark/shuffle/sort/ShuffleInMemorySorter.java

```diff
@@ -101,14 +94,17 @@ public void expandPointerArray(LongArray newArray) {
       array.getBaseOffset(),
       newArray.getBaseObject(),
       newArray.getBaseOffset(),
-      array.size() * (8 / memoryAllocationFactor)
+      pos * 8
```

Comment:
+1 on that. 
Is there any particular reason for removing the memoryAllocationFactor? Without this we might highly underutilize memory for TimSort.